p2p/metricshelper/registerer.go

```diff
@@ -0,0 +1,18 @@
+package metricshelper
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// RegisterCollectors registers the collectors with reg ignoring
+// reregistration error and panics on any other error
+func RegisterCollectors(reg prometheus.Registerer, collectors ...prometheus.Collector) {
+	for _, c := range collectors {
+		err := reg.Register(c)
+		if err != nil {
+			if _, ok := err.(prometheus.AlreadyRegisteredError); !ok {
```

Comment:
Probably slightly safer to do use `errors.Is`

Reply:
the returned error is an instance of the type 
```
type AlreadyRegisteredError struct {
	ExistingCollector, NewCollector Collector
}
```
Checking with errors.Is will not work since we do not know the existing collector, so I changed this to use errors.As

There is a tricky case here:
If an error is returned where we are registering a collector where a different collector with the same fully qualified name was registered before, this method will suppress it and the collector used as argument to the method will not be collected for metrics. In this example:
```
	reg := prometheus.NewRegistry()
	c1 := prometheus.NewCounter(
		prometheus.CounterOpts{
			Name: "counter",
		},
	)
	c2 := prometheus.NewCounter(
		prometheus.CounterOpts{
			Name: "counter",
		},
	)
	RegisterCollectors(reg, c1)
    RegisterCollectors(reg, c2) // c2 is not registered
    c1.Inc()
    c2.Inc() // This will have no effect
```
I'm leaving it as it is because this situation doesn't arise anywhere in the code yet since we are using namespaces for our metrics